src/Umbraco.PublishedCache.NuCache/Property.cs

```diff
@@ -382,9 +361,9 @@ private class CacheValue
 
     private class CacheValues : CacheValue
     {
-        private Dictionary<CompositeStringStringKey, CacheValue>? _values;
+        private readonly object _locko = new();
+        private ConcurrentDictionary<CompositeStringStringKey, CacheValue>? _values;
```

Comment:
```suggestion
        private ConcurrentDictionary<CompositeStringStringKey, CacheValue>? _values = InitializeConcurrentDictionary<CompositeStringStringKey, CacheValue>();
```

Reply:
Hi @Nuklon,

We shouldn't allocate the dictionary up front. It does not need allocating for non-variant properties:

![image](https://github.com/umbraco/Umbraco-CMS/assets/7405322/b1ac5cfb-8ded-4253-a473-826b59474640)

Reply:
> Hi @Nuklon,
> 
> We shouldn't allocate the dictionary up front. It does not need allocating for non-variant properties:
> 
> ![image](https://private-user-images.githubusercontent.com/7405322/322396030-b1ac5cfb-8ded-4253-a473-826b59474640.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMxNzYyMDAsIm5iZiI6MTcxMzE3NTkwMCwicGF0aCI6Ii83NDA1MzIyLzMyMjM5NjAzMC1iMWFjNWNmYi04ZGVkLTQyNTMtYTQ3My04MjZiNTk0NzQ2NDAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTVUMTAxMTQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjQzNjFiNzdhMmFhZDllMGJlY2RmYjEyMGUwYWNjNDZjN2M3MTMzNDI2NTRmZDRhODUwMzcxNzRmNTE5N2ZmZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.9QNfEOXN5hfoM8dvnGB3v0d_NT7u0hqm96pXaCkll5o)

Hmm, I see. That can be checked with `_sourceValueIsInvariant` I think? If you add ctor with isInvariant parameter, and create dictionary then if necessary?